File: bench/python/03-collections-iteration/05-expressive-collections.py
values = [4, -2, 7, 1]
summary = [min(values), max(values), sum(values)]
weighted_total = sum(index * value for index, value in enumerate([5, 10, 20]))
dot_product = sum(left * right for left, right in zip([1, 2, 3], [4, 5]))
until_negative = 0
for value in [2, 0, 5, -1, 100]:
    if value < 0:
        break
    until_negative += value

print(" | ".join("alpha,beta,gamma".split(",")))
print(summary)
print(weighted_total)
print(dot_product)
print(until_negative)

File: bench/rust/03-collections-iteration/05-expressive-collections.rs
fn main() {
    let values = [4, -2, 7, 1];
    let summary = [
        *values.iter().min().unwrap(),
        *values.iter().max().unwrap(),
        values.iter().sum(),
    ];
    let weighted_total: i32 = [5, 10, 20]
        .iter()
        .enumerate()
        .map(|(index, value)| index as i32 * value)
        .sum();
    let dot_product: i32 = [1, 2, 3]
        .iter()
        .zip([4, 5])
        .map(|(left, right)| left * right)
        .sum();
    let until_negative: i32 = [2, 0, 5, -1, 100]
        .iter()
        .take_while(|value| **value >= 0)
        .sum();

    println!("{}", "alpha,beta,gamma".split(',').collect::<Vec<_>>().join(" | "));
    println!("{summary:?}");
    println!("{weighted_total}");
    println!("{dot_product}");
    println!("{until_negative}");
}
